sycl/source/detail/graph_impl.cpp

```diff
@@ -152,14 +152,118 @@ bool graph_impl::clear_queues() {
   return AnyQueuesCleared;
 }
 
-void node_impl::exec(const std::shared_ptr<sycl::detail::queue_impl> &Queue
-                         _CODELOCPARAMDEF(&CodeLoc)) {
-  std::vector<sycl::event> Deps;
-  for (auto Sender : MPredecessors)
-    Deps.push_back(Sender->get_event());
+void exec_graph_impl::create_pi_command_buffers(sycl::device D,
+                                                const sycl::context &Ctx) {
+  // TODO we only have a single command-buffer per graph here, but
+  // this will need to be multiple command-buffers for non-trivial graphs
+  pi_ext_command_buffer OutCommandBuffer;
+  pi_ext_command_buffer_desc Desc{};
+  auto ContextImpl = sycl::detail::getSyclObjImpl(Ctx);
+  const sycl::detail::plugin &Plugin = ContextImpl->getPlugin();
+  auto DeviceImpl = sycl::detail::getSyclObjImpl(D);
+  pi_result Res =
+      Plugin.call_nocheck<sycl::detail::PiApiKind::piextCommandBufferCreate>(
+          ContextImpl->getHandleRef(), DeviceImpl->getHandleRef(), &Desc,
+          &OutCommandBuffer);
+  if (Res != pi_result::PI_SUCCESS) {
+    throw sycl::exception(errc::invalid, "Failed to create PI command-buffer");
+  }
+
+  MPiCommandBuffers[D] = OutCommandBuffer;
+
+  // TODO extract logic from enqueueImpKernel
```

Comment:
Looking at `enqueueImpKernel` really we're missing the kernel bundle logical, so this could be more precise and say "TODO extract kernel bundle logica from enqueueImpKernel"